PurchasesTests/Mocks/MockProductsManager.swift

```diff
@@ -40,35 +40,6 @@ class MockProductsManager: ProductsManager {
         }
     }
 
-    var invokedProductsFromOptimalStoreKitVersion = false
-    var invokedProductsFromOptimalStoreKitVersionCount = 0
-    var invokedProductsFromOptimalStoreKitVersionParameters: (identifiers: Set<String>, Void)?
-    var invokedProductsFromOptimalStoreKitVersionParametersList = [(identifiers: Set<String>, Void)]()
-
-    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    override func productsFromOptimalStoreKitVersion(withIdentifiers identifiers: Set<String>) async -> Set<ProductDetails> {
-        invokedProductsFromOptimalStoreKitVersion = true
-        invokedProductsFromOptimalStoreKitVersionCount += 1
-        invokedProductsFromOptimalStoreKitVersionParameters = (identifiers, ())
-        invokedProductsFromOptimalStoreKitVersionParametersList.append((identifiers, ()))
-        let result = stubbedProductsFromOptimalStoreKitVersionWithIdentifiersCompletionResult?.0 ?? Set<ProductDetails>()
-        return result
-    }
-
-    var invokedSk2ProductDetails = false
-    var invokedSk2ProductDetailsCount = 0
-    var invokedSk2ProductDetailsParameters: (identifiers: Set<String>, Void)?
-    var invokedSk2ProductDetailsParametersList = [(identifiers: Set<String>, Void)]()
-
-    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    override func sk2ProductDetails(withIdentifiers identifiers: Set<String>) async -> Set<SK2ProductDetails> {
-        invokedSk2ProductDetails = true
-        invokedSk2ProductDetailsCount += 1
-        invokedSk2ProductDetailsParameters = (identifiers, ())
-        invokedSk2ProductDetailsParametersList.append((identifiers, ()))
-        return Set()
-    }
```

Comment:
I guess this would open a conversation... For SK2 tests that require the use of StoreKit Test, do we mock the rest of the objects involved in the test? Or do we use real objects + StoreKitTest 

To give an example of this. If we don't use mocks and we use real implementations + StoreKit test, this function https://github.com/RevenueCat/purchases-ios/pull/863/files?file-filters%5B%5D=.pbxproj&file-filters%5B%5D=.swift&hide-deleted-files=true#diff-481adf9b3d275b768848e881f190b9de06af7f94ba7e4c864187aa0cc3225420R45 would be something like:

```
func setUpSK2WithError() throws {
        testSession = try SKTestSession(configurationFileNamed: "UnitTestsConfiguration")
        testSession.disableDialogs = true
        testSession.clearTransactions()

        userDefaults = UserDefaults(suiteName: "TestDefaults")

        let operationDispatcher = OperationDispatcher()
        let receiptFetcher = ReceiptFetcher(requestFetcher: StoreKitRequestFetcher(operationDispatcher: operationDispatcher))
        let introEligibilityCalculator = IntroEligibilityCalculator(productsManager: ProductsManager(productsRequestFactory: ProductsRequestFactory()), receiptParser: ReceiptParser())
        let systemInfo = try SystemInfo(platformFlavor: "iOS",
                                           platformFlavorVersion: "foo",
                                           finishTransactions: false)
        let httpClient = HTTPClient(systemInfo: systemInfo, eTagManager: ETagManager(userDefaults: userDefaults))
        let backend = Backend(httpClient: httpClient, apiKey: "api_key")
        let deviceCache = DeviceCache(userDefaults: userDefaults)
        let purchaserInfoManager = PurchaserInfoManager(operationDispatcher: operationDispatcher,
                                                        deviceCache: deviceCache,
                                                        backend: backend,
                                                        systemInfo: systemInfo)
        let identityManager = IdentityManager(deviceCache: deviceCache,
                                              backend: backend,
                                              purchaserInfoManager: purchaserInfoManager)
        let productsManager = ProductsManager()
        trialOrIntroPriceEligibilityChecker =
        TrialOrIntroPriceEligibilityChecker(receiptFetcher: receiptFetcher,
                                            introEligibilityCalculator: introEligibilityCalculator,
                                            backend: backend,
                                            identityManager: identityManager,
                                            operationDispatcher: operationDispatcher,
                                            productsManager: productsManager)
}
```
    
If we device to mock everything but the functions that we want to use StoreKitTest on, we would have to remove these two functions, as I did in this change https://github.com/RevenueCat/purchases-ios/pull/863/files?file-filters%5B%5D=.pbxproj&file-filters%5B%5D=.swift&hide-deleted-files=true#diff-a4729ae157b83ce077b44506492f3c4b8f7272dda6098c8cbf39ea971e775442L43

Reply:
I'd say use your best judgement. We're in a weird scenario where these aren't fully integration tests or fully unit tests. 
I'd go for mocking everything other than SK2, or maybe even further mocking everything that's in our scope and / or is easily mockable. 

in the BackendIntegrationTests bundle, though, I'd go for mocking as little as possible. Maybe using spies instead of mocks so that we can test stuff. 

What do you think?

Reply:
I like the idea of the `PartialMockProductsManager`. I think that's more clear